* Tue Mar 13 2007 Adam Jackson <ajax@redhat.com> 1.2.99.901-2
- xserver-1.2.99.901-xephyr-crash-at-exit.patch: Fix yet another GLX visual
  mess. (#231425)

https://bugs.freedesktop.org/show_bug.cgi?id=10713
http://cvs.fedora.redhat.com/viewcvs/devel/xorg-x11-server/xserver-1.2.99.901-xephyr-crash-at-exit.patch?rev=1.1&view=log
Debian bug #420421 (and maybe #424517)

--- xorg-server-1.2.99.901/GL/mesa/X/xf86glx.c.jx	2007-03-07 14:15:25.000000000 -0500
+++ xorg-server-1.2.99.901/GL/mesa/X/xf86glx.c	2007-03-13 14:27:38.000000000 -0400
@@ -281,12 +281,14 @@
     __GLXMESAscreen *mesaScreen = (__GLXMESAscreen *) screen;
     int i;
 
-    for (i = 0; i < mesaScreen->num_vis; i++) {
-	if (mesaScreen->xm_vis[i])
-	    XMesaDestroyVisual(mesaScreen->xm_vis[i]);
-    }
+    if (mesaScreen->xm_vis) {
+	for (i = 0; i < mesaScreen->num_vis; i++) {
+	    if (mesaScreen->xm_vis[i])
+		XMesaDestroyVisual(mesaScreen->xm_vis[i]);
+	}
 
-    xfree(mesaScreen->xm_vis);
+	xfree(mesaScreen->xm_vis);
+    }
 
     __glXScreenDestroy(screen);
 
